www/include/Administration/performance/viewData.php

```diff
@@ -137,57 +138,60 @@
 $brk = new CentreonBroker($pearDB);
 
 if ((isset($inputs["o1"]) && $inputs["o1"]) || (isset($inputs["o2"]) && $inputs["o2"])) {
-    if ($inputs["o"] == "rg" && isset($inputs["select"])) {
-        $selected = $inputs["select"];
-        foreach ($selected as $key => $value) {
-            $DBRESULT = $pearDBO->query("UPDATE index_data SET `must_be_rebuild` = '1' WHERE id = '" . $key . "'");
+    //filter integer keys
+    $selected = array_filter(
+        $inputs["select"],
+        function ($k) {
+            if (is_int($k)) {
```

Comment:
I would have checked for int($k) !== 0 and $k > 0 and returned (int) $k.
To avoid to remove all quoted $k like '12'